src/parsers/manifest/dash/indexes/base.ts

```diff
@@ -165,16 +170,26 @@ export default class BaseRepresentationIndex implements IRepresentationIndex {
                                     representationId,
                                     representationBitrate);
 
-    // TODO If indexRange is either undefined or behind the initialization segment
+    // TODO If indexRange is behind the initialization segment
     // the following logic will not work.
-    // However taking the nth first bytes like `dash.js` does (where n = 1500) is
-    // not straightforward as we would need to clean-up the segment after that.
-    // The following logic corresponds to 100% of tested cases, so good enough for
-    // now.
-    const range : [number, number] | undefined =
-      index.initialization !== undefined ? index.initialization.range :
-      index.indexRange !== undefined ? [0, index.indexRange[0] - 1] :
-                                       undefined;
+    // If no range and index range are given by manifest, we take
+    // as init segment the nth first bytes (where n = 1500).
+    // Therefore, we need to filter on init boxes after the segment
+    // is loaded, to ensure that we push a complete and dry segment
+    // to buffers.
+    let range: [number, number] | undefined;
+    this._shouldGuessInitRange = false;
+    if (index.initialization === undefined &&
+        index.indexRange === undefined &&
+        this._isContentFragmented) {
+      this._shouldGuessInitRange = true;
```

Comment:
I don't get the point of this variable either.
We could just use a specific logic in getInitSegment for when we do have an init segment but do not know its range no?